chainer/functions/activation/softmax.py

```diff
@@ -16,8 +16,9 @@ class Softmax(function.Function):
 
     """Softmax activation function."""
 
-    def __init__(self, use_cudnn=True):
+    def __init__(self, use_cudnn=True, axis=1):
         self.use_cudnn = use_cudnn
+        self.axis = axis
 
     def check_type_forward(self, in_types):
         type_check.expect(in_types.size() == 1)
```

Comment:
How about adding a new type checking at L:30 to ensure the `self.axis` is smaller than `x_type.ndim`?